SSL example needs crt and key file creation instructions

The example has a link on this line:

https://github.com/zio/zio-http/blob/1e1be6950475987bd43049bc121e24a0176952d1/docs/examples/basic/https-server.md?plain=1#L23

to instructions for creating certificate. But the instructions do not create a crt and key file. They produce a java keystore and cert file. And there is something about importing, which should not be necessary.

Attempting to follow those instructions and find other internet resources to fill in the missing steps to produce a crt and key file, and follow the example for configuration of ZIO HTTP have resulted in non-working server.

This is the error:

```
timestamp=2023-05-30T12:50:13.689708Z level=WARN thread=#zio-fiber-81 message="Fatal exception in Netty" cause="Exception in thread "zio-fiber-" io.netty.handler.codec.DecoderException: java.lang.NullPointerException: keyCertChain
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:499)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
        at io.netty.channel.kqueue.AbstractKQueueStreamChannel$KQueueStreamUnsafe.readReady(AbstractKQueueStreamChannel.java:544)
        at io.netty.channel.kqueue.AbstractKQueueChannel$AbstractKQueueUnsafe.readReady(AbstractKQueueChannel.java:387)
        at io.netty.channel.kqueue.KQueueEventLoop.processReady(KQueueEventLoop.java:213)
        at io.netty.channel.kqueue.KQueueEventLoop.run(KQueueEventLoop.java:291)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
        Suppressed: java.lang.NullPointerException: keyCertChain
                at io.netty.util.internal.ObjectUtil.checkNotNull(ObjectUtil.java:39)
                at io.netty.util.internal.ObjectUtil.checkNonEmpty(ObjectUtil.java:211)
                at io.netty.handler.ssl.SslContextBuilder.keyManager(SslContextBuilder.java:436)
                at io.netty.handler.ssl.SslContextBuilder.keyManager(SslContextBuilder.java:422)
                at io.netty.handler.ssl.SslContextBuilder.keyManager(SslContextBuilder.java:345)
                at io.netty.handler.ssl.SslContextBuilder.forServer(SslContextBuilder.java:84)
                at zio.http.netty.server.SSLUtil$.sslConfigToSslContext(ServerSSLDecoder.scala:76)
                at zio.http.netty.server.ServerSSLDecoder.decode(ServerSSLDecoder.scala:86)
                at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:529)
                at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468)
                at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
                at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
                at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
                at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
                at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
                at io.netty.channel.kqueue.AbstractKQueueStreamChannel$KQueueStreamUnsafe.readReady(AbstractKQueueStreamChannel.java:544)
                at io.netty.channel.kqueue.AbstractKQueueChannel$AbstractKQueueUnsafe.readReady(AbstractKQueueChannel.java:387)
                at io.netty.channel.kqueue.KQueueEventLoop.processReady(KQueueEventLoop.java:213)
                at io.netty.channel.kqueue.KQueueEventLoop.run(KQueueEventLoop.java:291)
                at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
                at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
                at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
                at java.base/java.lang.Thread.run(Thread.java:833)"
May 30, 2023 5:50:13 AM io.netty.channel.DefaultChannelPipeline onUnhandledInboundException
WARNING: An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.handler.codec.DecoderException: java.lang.NullPointerException: keyCertChain
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:499)
        at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:290)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412)
        at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440)
        at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919)
        at io.netty.channel.kqueue.AbstractKQueueStreamChannel$KQueueStreamUnsafe.readReady(AbstractKQueueStreamChannel.java:544)
        at io.netty.channel.kqueue.AbstractKQueueChannel$AbstractKQueueUnsafe.readReady(AbstractKQueueChannel.java:387)
        at io.netty.channel.kqueue.KQueueEventLoop.processReady(KQueueEventLoop.java:213)
        at io.netty.channel.kqueue.KQueueEventLoop.run(KQueueEventLoop.java:291)
        at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
        at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.NullPointerException: keyCertChain
        at io.netty.util.internal.ObjectUtil.checkNotNull(ObjectUtil.java:39)
        at io.netty.util.internal.ObjectUtil.checkNonEmpty(ObjectUtil.java:211)
        at io.netty.handler.ssl.SslContextBuilder.keyManager(SslContextBuilder.java:436)
        at io.netty.handler.ssl.SslContextBuilder.keyManager(SslContextBuilder.java:422)
        at io.netty.handler.ssl.SslContextBuilder.keyManager(SslContextBuilder.java:345)
        at io.netty.handler.ssl.SslContextBuilder.forServer(SslContextBuilder.java:84)
        at zio.http.netty.server.SSLUtil$.sslConfigToSslContext(ServerSSLDecoder.scala:76)
        at zio.http.netty.server.ServerSSLDecoder.decode(ServerSSLDecoder.scala:86)
        at io.netty.handler.codec.ByteToMessageDecoder.decodeRemovalReentryProtection(ByteToMessageDecoder.java:529)
        at io.netty.handler.codec.ByteToMessageDecoder.callDecode(ByteToMessageDecoder.java:468)
        ... 16 more
```

I've tried a technique I used elsewhere using openssl command:

```
openssl req -x509 -out server.crt -keyout server.key \
  -newkey rsa:2048 -nodes -sha256 \
  -subj '/CN=localhost' -extensions EXT -config <( \
   printf "[dn]\nCN=localhost\n[req]\ndistinguished_name = dn\n[EXT]\nsubjectAltName=DNS:localhost\nkeyUsage=digitalSignature\nextendedKeyUsage=serverAuth")
```

and a combination of keytool and openssl to start in a similar way to the referenced medium article but end up with crt and key files:

```
keytool -genkeypair -keyalg RSA -alias server -keystore server.jks -storepass 123456 -dname CN=localhost
keytool -importkeystore -srckeystore server.jks -srcstorepass 123456 -destkeystore server.p12 -srcstoretype jks -deststoretype pkcs12 -deststorepass 123456
openssl pkcs12 -in server.p12 -passin pass:123456 -out server.pem -passout pass:123456
openssl x509 -outform der -in server.pem -passin pass:123456 -out server.crt
openssl rsa -in server.pem -passin pass:123456 -out server.key
```

(This option generates a binary crt file which is different from what the pure openssl one does.)

--- a/docs/examples/basic/https-server.md
+++ b/docs/examples/basic/https-server.md
@@ -16,11 +16,18 @@ object HttpsHelloWorld extends ZIOAppDefault {
   }
 
   /**
-   * In this example an inbuilt API using keystore is used. For testing this
-   * example using curl, setup the certificate named "server.crt" from resources
-   * for the OS. Alternatively you can create the keystore and certificate using
-   * the following link
-   * https://medium.com/@maanadev/netty-with-https-tls-9bf699e07f01
+   * In this example, a private key and certificate are loaded from resources.
+   * For testing this example with curl, make sure the private key "server.key",
+   * and the certificate "server.crt" are inside the resources directory,
+   * which is by default "src/main/resources".
+   *
+   * You can use the following command to create a self-signed TLS certificate.
+   * This command will create two files: "server.key" and "server.crt".
+   *
+   *     openssl req -x509 -newkey rsa:4096 -sha256 -days 365 -nodes \
+   *       -keyout server.key -out server.crt \
+   *       -subj "/CN=example.com/OU=?/O=?/L=?/ST=?/C=??" \
+   *       -addext "subjectAltName=DNS:example.com,DNS:www.example.com,IP:10.0.0.1"
    */
 
   val sslConfig = SSLConfig.fromResource(

